### PR TITLE
This competition will be huuuuuuggggeeee

### PR DESCRIPTION
**Pull request by BeastRemus**

### CODE REVIEW

**Review by SecondThread — APPROVED** [review]
Looks amazing; everything seems to be well tested. Yeet!

File: src/auto/commands/DriveAtAngle.java [hunk 1]
```diff
@@ -17,46 +19,120 @@
 	private double slowSpeed, fastSpeed;
 	private double angle;
 	private AHRS navx;
-	
+	private PIDController controller;
```
Comment (tomlazar) [inline src/auto/commands/DriveAtAngle.java:22]: This class should be called `DriveAtAngleCommand`

File: src/auto/commands/DriveAtAngle.java [hunk 2]
```diff
@@ -17,46 +19,120 @@
...
+	 * 
+	 * @param stop
+	 * @param speed
+	 * @param angle
+	 */
+	public DriveAtAngle(IStopCondition stop, double speed, double angle) {
+		usePID = true;
+		// these will most likely be small as the value needs to be under 1.0/
+		// -1.0
+		controller = new PIDController(0.01, 0.00005, .0, true, false);
```
Comment (tomlazar) [inline src/auto/commands/DriveAtAngle.java:36]: Extract pid values to named constants

File: src/auto/commands/DrivingPegVisionCommand.java [hunk 3]
```diff
@@ -2,109 +2,104 @@
...
 public class DrivingPegVisionCommand implements ICommand {
 	private NetworkTables table;
 	private double degreesToTurn;
 	private double distanceToMove;
-	private boolean doneWithVision = false;
-	private ArrayList<ICommand> subcommands = new ArrayList<>();
-	private double percentToFinish = 0;
+	private boolean doneWithVision=false;
+	private CommandListRunner runner=null;
+	private double percentToFinish=0;
```
Comment (tomlazar) [inline src/auto/commands/DrivingPegVisionCommand.java:25]: these values should be initialized in init commands not at declaration

File: src/babbage/BabbageAutoLists.java [hunk 4]
```diff
@@ -29,13 +32,11 @@
...
 	public static CommandList centerRed(){
-		
 		encoders.add((Encoder) sensors.getSensor("ENCODER0"));
-		
 		CommandList CL = new CommandList();
-		SmartWriter.putS("AutoList", "centerRed", DebugMode.COMPETITION);
-		//CL.addCommand(new DriveCommand(new DistanceStopCondition(encoders, distanceFromWall), speed));
-		CL.addCommand(new RunPegVisionCommand(1));
+		CL.addCommand(new WaitCommand(new TimerStopCondition(500)));
+		CL.addCommand(new ContinuousPegVisionCommand(1));
```
Comment (tomlazar) [inline src/babbage/BabbageAutoLists.java:38]: why does continuousPegVisionCommand Not have a proper wait command?

File: src/babbage/HighGoalTurning.java [hunk 5]
```diff
@@ -12,17 +12,18 @@
 
 public class HighGoalTurning extends IControl {
 
-	private ServoMotor servo;
+	private ServoMotor servo, heightServoMotor;
 	private NetworkTables table;
 	private boolean waitingToTurnShooter=false;
 	private double targetAngle=90;
 	//private XboxController controller;
 	private boolean processingVision=false;
```
Comment (tomlazar) [inline src/babbage/HighGoalTurning.java:20]: these should all be in final or initialized in init commands

File: src/physicalOutput/TalonSRX.java [hunk 6]
```diff
@@ -24,12 +25,13 @@ public TalonSRX(int x, boolean reverse, boolean hasEncoder) {
...
 			part.configEncoderCodesPerRev(4096);
 			part.setProfile(0);
-			part.setF(0.013); /// 0.008562
-			part.setP(0.018); // 102/4000 .1*1023/4000Uerror = 0.0255
-			part.setI(0.00002); // set P/100 to start
+			part.setF(0.001); /// 0.008562
+			part.setP(0.08); // 102/4000 .1*1023/4000Uerror = 0.0255
+			part.setI(0.0001); // set P/100 to start
 			part.setD(0.0000);
 			part.setAllowableClosedLoopErr(0);
```
Comment (tomlazar) [inline src/physicalOutput/TalonSRX.java:32]: Extract magical numbers to named constants

**Pull request merged**